### PR TITLE
Complete refactoring of mkpydeps

### PR DESCRIPTION
**Pull request by robsonpeixoto**
- Add dependency resolution using topological sort
  - Using jinja to print the template
  - Print a complete Formula template

### CODE REVIEW

**Comment by robsonpeixoto** [comment]
I closed #1 and #2 in favor of this PR

**Comment by tdsmith** [comment]
Fancy. :o

This has juuuuuuust enough complexity that I'll probably spin this out into a new repository and actually make it a proper pypi package, heh. I appreciate the work! Stay tuned!

**Comment by robsonpeixoto** [comment]
Thanks. I don't known how put it in a separated package. I created the function above on my ~/.bashrc to create a empty environment, install the package how you like to create the Formula, install the script requirement and run the script.

But it can fail the one of the packages install has incomum requirements with different versions.

``` bash
mkpydeps() {
  if [[ -z $1 ]]; then
    echo "Pass a package as parameter"
    return 1
  fi
  CWD=$(pwd)
  mktmpenv
  pip install $1
  pip install tl.eggdeps jinja2 networkx
  $HOME/src/labmisc/mkpydeps $1
  deactivate
  cd $CWD
}
```

**Comment by robsonpeixoto** [comment]
Do you known any library

**Comment by tdsmith** [comment]
(I'll get to this tomorrow, promise!)

**Comment by robsonpeixoto** [comment]
Relex @tdsmith =D
Do it in your time

File: mkpydeps [hunk 1]
```diff
@@ -7,37 +7,94 @@ installed in sys.path**, determines which packages foo and its dependents
...
-print '        system "python", *Language::Python.setup_install_args(libexec/"vendor")'
-print '      end'
-print '    end'
+            G.node[n]['url'] = i['url']
+            f = urllib2.urlopen(G.node[n]['url'])
+            G.node[n]['checksum'] = sha1(f.read()).hexdigest()
+
+# get the dependency resolution order
+deps = nx.algorithms.dag.topological_sort(G)
+deps.reverse()
```
Comment (tdsmith) [inline mkpydeps:91]: Hey, you never use this :p

Comment (robsonpeixoto) [inline mkpydeps:91]: This method do not return anything. It's a side effect in the list:

``` python
>>> l = [1,2,3,4,5]
>>> lr = l.reverse()
>>> lr
>>> l
[5, 4, 3, 2, 1]
```

**Comment by robsonpeixoto** [comment]
Thanks

File: mkpydeps (hunk 1 above)
Comment (tdsmith) [inline mkpydeps:91]: But `deps` is never used.

Comment (robsonpeixoto) [inline mkpydeps:91]: Good catch. Sorry

**Pull request closed without merging**